tests/integration-tests/tests/create/test_create.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import pytest
+from remote_command_executor import RemoteCommandExecutor
+
+from tests.common.assertions import assert_errors_in_logs
+from tests.common.utils import retrieve_latest_ami
+
+
+@pytest.mark.dimensions("eu-central-1", "c5.xlarge", "ubuntu1804", "slurm")
+def test_create_wrong_os(region, os, pcluster_config_reader, clusters_factory, test_datadir, scheduler, instance):
+    """Test error message when os provide is different from the os of custom AMI"""
+    # ubuntu1804 is specified in the config file but an AMI of centos7 is provided
+    wrong_os = "centos7"
```

Comment:
We would want to do a quick assertion that wrong_os != provided os here, in case we change the dimensions by accident in future